File: algorithm-leetcode-jvm/src/test/java/org/muxun/algorithm/leetcode/java/greedy/LeetCode1702Test.java
package org.muxun.algorithm.leetcode.java.greedy;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

/**
 * @author 慕勋
 * @created 2024-04-10
 */
@DisplayName("修改后的最大二进制字符串测试用例")
public class LeetCode1702Test {

    @Test
    public void testOne() {
        String binary = "000110";
        String expected = "111011";
        String res = new LeetCode1702().maximumBinaryString(binary);
        Assertions.assertEquals(expected, res);
    }

    @Test
    public void testTwo() {
        String binary = "01";
        String expected = "01";
        String res = new LeetCode1702().maximumBinaryString(binary);
        Assertions.assertEquals(expected, res);
    }
}

File: algorithm-leetcode-jvm/src/test/kotlin/org/muxun/algorithm/leetcode/kotlin/greedy/LeetCode1702Test.kt
package org.muxun.algorithm.leetcode.kotlin.greedy

import kotlin.test.Test
import kotlin.test.junit5.JUnit5Asserter.assertEquals

class LeetCode1702Test {

    @Test
    fun testOne() {
        val binary = "000110"
        val expected = "111011"
        val res = LeetCode1702().maximumBinaryString(binary)
        assertEquals("修改后的最大二进制字符串测试用例1", expected, res)
    }

    @Test
    fun testTwo() {
        val binary = "01"
        val expected = "01"
        val res = LeetCode1702().maximumBinaryString(binary)
        assertEquals("修改后的最大二进制字符串测试用例2", expected, res)
    }
}